core/domain/skill_domain.py

```diff
@@ -1776,3 +1776,72 @@ def from_dict(cls, skill_mastery_dict):
             skill_mastery_dict['skill_id'],
             skill_mastery_dict['degree_of_mastery']
         )
+
+
+class CategorizedSkills:
+    """Domain object for representing categorized skills' ids and
+    descriptions.
+    """
+
+    def __init__(self):
+        """Constructs a CategorizedSkills domain object."""
+        self.categorized_skills = {}
+
+    def add_topic(self, topic_name):
+        """Adds a topic to the categorized skills and initializes its value
+        as an empty dict.
+
+        Args:
+            topic_name: str. The name of the topic.
+        """
+        self.categorized_skills[topic_name] = {}
+
+    def initialize_uncategorized_skills(self, topic_name):
+        """Initializes the uncategorized skills for a topic as an empty list.
+
+        Args:
+            topic_name: str. The name of the topic.
+        """
+        self.categorized_skills[topic_name]['uncategorized'] = []
+
+    def initialize_subtopic_skills(self, topic_name, subtopic_title):
+        """Initializes the subtopic skills for a topic as an empty list.
+
+        Args:
+            topic_name: str. The name of the topic.
+            subtopic_title: str. The title of the subtopic.
+        """
+        self.categorized_skills[topic_name][subtopic_title] = []
+
+    def update_uncategorized_skills(
```

Comment:
Should be add_uncategorized_skill because that's what this method is actually doing.

Reply:
done.